---
output: github_document
---
# Progress Report

## Deviation from the final proposal.
As described in our final proposal, we are still using the supplemental file `GSE152075_raw_counts_GEO.txt` and `PhenoData` from the `GSE152075`as a source of the raw counts and metadata, respectively. 
Analysis of the aforementioned data sets for differential gene expression, we initially proposed to leverage DESeq2. However, we have opted for EdgeR because the authors of the paper used DESeq2 and based on the premise that no statistical modelling can fully capture all biological phenomena, we wanted to see if a different  pipeline of analysis would be comparable to what was originally found.
All group members contributed to the current progress of our project as described in the final proposal. 

## Progress of Analyses
The following steps describe the methodology used so far in our analyses:

1. Preliminary data exploration was done using `dplyr` functions. Using `ggplot2 `and `pheatmap` packages, boxplots and heatmaps were generated.
2. Using functions from the `EdgeR` package, data a `DGEList` was created amd the count data was normalized using the "TMM" method. 
3. The `decideTestsDGE()` function was used to calculate differentially expressed genes. 
4. From the list of up-regulated and down-regulated genes obtained from step 4, principal component analyses (PCA) was performed using functions from the `stat` package.
5. A logistic regression model was generated to predict infection status based on RNAseq data.

Based on the above steps, aims 1 and 3 of our final proposal were partially answered. For aim 1, we have characterized the differentially expressed genes based on the infection status, i.e. positive vs negative. As for aim 3, the infection status was predicted after sub-setting our PCA data into training and validation sets.
Other than the switch from DESeq2 to EdgeR for expression analyses, our goals as stipulated in our final proposal remain the same.

## Results
Differential gene expression was observed between individuals with negative and positive COVID infection status. 45 genes were up-regulated while 21 were down-regulated when sequencing batch was accounted for. When compared to what the authors of the paper found, we can determine that we have positive results though we have ~20 less genes than they got. This discrepancy can be attributed to the genes that we dropped based on our high CPM and sample thresholds.
The following is a table showing the upregulated genes:
```{r,echo=FALSE}
upreg <- read.delim("~/Desktop/GSAT 540/Group/project-team-11/Results/DE_Upregulated.tsv")
head(knitr::kable(upreg), 10)
```

The following is a table showing the downregulated genes:
```{r,echo=FALSE}
dnreg <- read.delim("~/Desktop/GSAT 540/Group/project-team-11/Results/DE_Downregulated.tsv")
head(knitr::kable(dnreg), 10)
```



 