---
title: "<big><big>Host transcriptional response to SARS-Cov2 by Sex and Age.</big></big>"
author: "<br><font size=6> Ekpereka Amutaigwe, Credo Casmil, Dollina Dodani, Aditi Nallan </font>"
date: "<font size=6> 30 March 2022"
output:
  xaringan::moon_reader:
    css: xaringan-themer.css
    self_contained: false
    lib_dir: "libs"
    nature:
      highlightStyle: github
      highlightLines: true
      countIncrementalSlides: false
      ratio: "16:9"
---


```{r, echo = FALSE}
knitr::opts_chunk$set(tidy = FALSE, tidy.opts=list(width.cutoff=80), fig.retina=3)
ggplot2::theme_set(ggplot2::theme_bw(base_size = 20))
ggplot2::update_geom_defaults("point", list(size = 3))
```

```{r xaringan-themer, include=FALSE, warning=FALSE}
library(xaringanthemer)
style_mono_accent(
  base_color = "#4E4D4F",
  header_font_google = google_font("Noto Serif Display"),
  text_font_google   = google_font("Nanum Myeongjo"),
  code_font_google   = google_font("Inconsolata"),
  base_font_size = "22px", 
  header_h1_font_size = "2rem",
  title_slide_background_size = "15%",
  title_slide_background_position = "95% 85%",
  link_color = "rgb(104, 27, 148)",
  link_decoration = "underline"
)
```

```{css, echo = FALSE}
pre {
  white-space: pre-wrap;
}
.remark-code {
  background: #f8f8f8;
}
.remark-inline-code {
  background: "white";
}
.remark-code {
  font-size: 22px;
}
.huge .remark-code { /*Change made here*/
  font-size: 200% !important;
}
.tiny .remark-code { /*Change made here*/
  font-size: 60% !important;
}
.smaller .remark-code { /*Change made here*/
  font-size: 90% !important;
}
.smaller2 .remark-code { /*Change made here*/
  font-size: 80% !important;
}
.smaller3 .remark-code { /*Change made here*/
  font-size: 70% !important;
}
```


# Background

<small>
- SARS-CoV-2, a novel coronavirus, has rapidly spread worldwide and has shown a wide range of clinical manifestations among different population groups.

- Clinically, COVID-19 cases tend to be more severe for older adults and males

- [Lieberman et al. (Plos Biology 2020)](https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.3000849) examined host response gene expression across infection status, viral load, age, and sex among shotgun RNA sequencing profiles of nasopharyngeal (NP) swabs from 430 individuals with PCR-confirmed SARS-CoV-2 and 54 negative controls.

- Their study described gene signatures associated with Viral load extensively but the same analysis for Age and Sex was lacking. 

</small>

```{r, echo=FALSE, out.width="70%", fig.show = 'hold', fig.cap = 'Figure from Lieberman et al. (Plos Biology 2020) paper', fig.align="center"}
knitr::include_graphics("Heatmap_paper.png")
```

---

## Project Aim

- Describe gene signatures associated with COVID infection by extending the original research to explore the interactive effects of age and sex on gene expression profiles.

- Leverage these signatures in a Machine Learning model to predict infection state.

---

```{r, echo = FALSE, warning = FALSE, message = FALSE}
# read in data behind the scenes
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(GEOquery))
suppressPackageStartupMessages(library(limma))
suppressPackageStartupMessages(library(edgeR))
suppressPackageStartupMessages(library(pheatmap))
suppressPackageStartupMessages(library(statmod))
suppressPackageStartupMessages(library(ggrepel))
suppressPackageStartupMessages(library(ggbiplot))
suppressPackageStartupMessages(library(ggplot2))
suppressPackageStartupMessages(library(GGally))
suppressPackageStartupMessages(library(caTools)) ## used for splitting the data into training and validation sets
suppressPackageStartupMessages(library(caret)) ## used for confusion matrix, accuracy, sensitivity, and specificity
suppressPackageStartupMessages(library(mice))
suppressPackageStartupMessages(library(VIM))
suppressPackageStartupMessages(library(sva))
suppressPackageStartupMessages(library(missForest))
suppressPackageStartupMessages(library(RANN))
suppressPackageStartupMessages(library(data.table))
suppressPackageStartupMessages(library(goseq))
suppressPackageStartupMessages(library(GOplot))
suppressPackageStartupMessages(library(clusterProfiler))
suppressPackageStartupMessages(library(org.Hs.eg.db))
suppressPackageStartupMessages(library(viridis))

#set working directory
setwd(getwd())

#data download
raw_counts = read.csv("../GSE152075/GSE152075_raw_counts_GEO.txt", sep = "")
c = as.matrix(raw_counts,row.names="gene_id")

#metadata
pdata = read.csv("../GSE152075/Metadata.tsv", header = TRUE, sep = "\t")
m = subset(pdata, select = c(1,39,40,41,42,43))

#Rename metadata columns
colnames(m) = c("Sample", "Age", "Gender", "Viral_load", "Sars_test", "Batch")
m <- within(m, Age[Age == "90+"] <- "90")

#Modify the subset data
m <- m %>%
  transform(Age = as.numeric(Age))
convert = c("Gender", "Sars_test", "Batch")
m[convert] <- lapply(m[convert], factor)
m$Viral_load[m$Viral_load == "N/A"] <- 0
m$Viral_load = as.numeric(m$Viral_load)
```

## Dataset

- Obtained from Lieberman et al. (Plos Biology 2020) paper. 

- This data was obtained in two separate files-

  - **Main data file**: expression values - one per gene, per sample ( $G$ genes $\times$ $N$ samples)
  
  - **Metadata file**: several covariates/experimental design variables for each sample ( $N$ samples $\times$ $P$ covariates). This file was modified to retain only the columns of interest that is- Sample, Age, Sex, Batch, Viral load and Sars test result.

```{r}
# Expression matrix
dim(c)
# Metadata
dim(m)
```

---

# Metadata 

```{r}
summary(m)
```

---

